pricelist_per_product/pricelist.py

```diff
@@ -91,12 +93,39 @@ def button_product(self):
         }
 
     @api.model
+    def update_sequence(self, vals):
+        if (vals.get('product_id') and vals.get('related_sequence') and
+                vals['related_sequence'] != 5):
+            vals['related_sequence'] = 5
+        elif (vals.get('product_tmpl_id') and vals.get('related_sequence') and
+              vals['related_sequence'] != 10):
+            vals['related_sequence'] = 10
+        return vals
+
+    @api.model
     def create(self, vals):
         if self.env['product.pricelist.version'].browse(
                 vals['price_version_id']).price_grid:
+            vals = self.update_sequence(vals)
             vals.update({
                 'price_discount': -1,
-                'sequence': 1,
                 'base': 1,
             })
         return super(ProductPricelistItem, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        for item in self:
+            values = {}
+            if (not vals.get('product_id') and item.product_id and
```

Comment:
it could be easier if you write: `product_id = vals.get('product_id', item.product_id)` instead to evaluate `if (not vals.get('product_id') and item.product_id ...`. Idem for product_tmpl_id and sequence
Then you can compare product_id and product_tmpl_id directly in their future values